lib/omnibus/packagers/deb.rb

```diff
@@ -676,33 +676,42 @@ def remove(packages)
       shellout!("apt-get remove -y --force-yes #{packages}")
     end
 
-    def compression_level(val = nil)
-      unless val.nil?
+    def compression_level(val = NULL)
+      if null?(val)
+        @compression_level || 6
```

Comment:
Pardon my lack of ruby knowledge, not sure what the `||` does. Would `=` be more meaningful?
```suggestion
        @compression_level = 6
```

Reply:
We're in the same boat, I don't understand ruby very much :sweat_smile: 

My understanding is that this function is a getter & setter. It will behave as a getter when the passed value is `NULL` (and/or `nil`? but other functions in the file seem to be using `NULL`)

So in this branch, we're returning a value, which is either `@compression_level` if it was set, or `6` as a default value.

The setter part is in the other branch, when `val` isn't `NULL`, in which case we assign it to `@compression_level`

Reply:
After more testing and explanations from Alex, I switched back to nil, but have to stick to `@foo || default`, otherwise we assign the default value back everytime we fetch it, which defeats the purpose of having a setter